nautobot/dcim/api/serializers.py

```diff
@@ -145,17 +148,18 @@ def get_connected_endpoint(self, obj):
         """
         Return the appropriate serializer for the type of connected object.
         """
-        if obj._path is not None and obj._path.destination is not None:
-            depth = get_nested_serializer_depth(self)
-            return return_nested_serializer_data_based_on_depth(
-                self, depth, obj, obj._path.destination, "connected_endpoint"
-            )
+        with contextlib.suppress(CablePath.DoesNotExist):
+            if obj._path is not None and obj._path.destination is not None:
+                serializer = get_serializer_for_model(obj._path.destination, prefix="Nested")
+                context = {"request": self.context["request"]}
+                return serializer(obj._path.destination, context=context).data
```

Comment:
Changes here look suspect - please double-check.

Reply:
Sorry for the oversight when rebasing, should be fixed now.